colossus/src/main/scala/colossus/protocols/http/HttpResponse.scala

```diff
@@ -1,77 +1,151 @@
 package colossus
 package protocols.http
 
-import akka.util.ByteString
-import colossus.controller.{FiniteBytePipe, Source}
+import akka.util.{ByteString, ByteStringBuilder}
+import colossus.controller.Source
 import colossus.core.DataBuffer
+import core._
+import service.Callback
+import controller._
+import HttpParse._
+
+
+case class HttpResponseHead(version : HttpVersion, code : HttpCode, headers : Vector[(String, String)] = Vector()) extends HttpHeaderUtils {
```

Comment:
Not a comment for this PR, but in general, I've been increasingly noticing our headers aren't a case class, but a tuple(which is essentially a case class).  SHould we just make it a case class and have nice names on it?